docs-chef-io/content/inspec/resources/parse_config_file.md

```diff
@@ -145,3 +126,35 @@ Use `standalone_comments: false`, to parse the following:
 
     'key = value # comment'
     params['key'] = 'value'
+
+## Matchers
+
+For a full list of available matchers, please visit our [matchers page](/inspec/matchers/).
+
+## Examples
+
+The following examples show how to use this Chef InSpec audit resource.
+
+### Test a configuration setting
+
+    describe parse_config_file('/path/to/file.conf') do
+     its('PARAM_X') { should eq 'Y' }
+    end
+
+### Use options, and then test a configuration setting
+
+    describe parse_config_file('/path/to/file.conf', { multiple_values: true }) do
+     its('PARAM_X') { should include 'Y' }
+    end
+
+### Test a file with an ini-like structure (such as a yum.conf)
+
+    describe parse_config_file('/path/to/yum.conf') do
+     its('main') { should include('gpgcheck' => '1') }
+    end
+
+### Test a configuration setting containing dots
+
+    describe parse_config_file('/etc/sysctl.conf') do
+     its(['kernel.domainname']) { should eq 'example.com' }
```

Comment:
This is just opinion.

This just feels strange - I think this may be a case of implementation driving UX. If I were a user, and I wanted to do a query containing a dot, even assuming I knew it had to be escaped, the typical thing would be to use a slash, not to enclose the query in brackets.

I agree the need is valid, and here we wish we could hijack `rspec-its` dot-processing facility entirely. So why not do so? In method_missing, why not explicitly search for dots and transform the call as needed to fool `rspec-its`? Then there may be no need to alter the DSL syntax at all - you may be able to have:

```
its('kernel.domainname') { should eq 'example.com' }
```

simply work.

Reply:
From my understanding of rspec-its' code our method_missing won't ever see the dot due to the code at:
https://github.com/rspec/rspec-its/blob/v1.2.0/lib/rspec/its.rb#L113

I drafted some code to return some magic object from method_missing so that the calls to `send` Just Work.
It fails, however, whenever the hash contains "foo.bar" and "foo.bar.baz" and one wants to get the latter.

If that sounds acceptable I can use that instead of accepting arrays (like other resources do already).

I don't know how it should behave with regards to_sth methods. E.g. only "foo.bar.baz" exists, but the caller wants 'foo.bar' only.
On the plus side, `its('foo.bar.baz.downcase')` just works.

Reply:
Ah, thank you, that code reference cleared up the situation for me. I think your existing approach using the array syntax is an acceptable compromise.

Reply:
If it's okay for you to leave it as-is then I think this pull request is ready from my POV.